api/server/sdk/server.go

```diff
@@ -83,14 +124,111 @@ type Server struct {
 }
 
 // Interface check
-var _ grpcserver.Server = &Server{}
+var _ grpcserver.Server = &sdkGrpcServer{}
 
-// New creates a new SDK gRPC server
+// New creates a new SDK server
 func New(config *ServerConfig) (*Server, error) {
+
+	if config == nil {
+		return nil, fmt.Errorf("Must provide configuration")
+	}
+
+	// Check if the socket is provided to enable the REST gateway to communicate
+	// to the unix domain socket
+	if len(config.Socket) == 0 {
+		return nil, fmt.Errorf("Must provide unix domain socket for SDK")
+	}
+
+	// Create a gRPC server on the network
+	netServer, err := newSdkGrpcServer(config)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a gRPC server on a unix domain socket
+	udsConfig := *config
+	udsConfig.Net = "unix"
+	udsConfig.Address = config.Socket
+	udsConfig.Tls = nil
+	udsServer, err := newSdkGrpcServer(&udsConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create REST Gateway and connect it to the unix domain socket server
+	restGateway, err := newSdkRestGateway(config, udsServer)
```

Comment:
we should error out if config.RestPort (required by rest gateway) is not provided.

Reply:
Good point